src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -66,25 +66,26 @@ public VisualStudioProject CreateAndAddToWorkspace(string projectSystemName, str
                         compilationOptions: creationInfo.CompilationOptions,
                         parseOptions: creationInfo.ParseOptions);
 
-                // HACK: update this since we're still on the UI thread. Note we can only update this if we don't have projects -- the workspace
-                // only lets us really do this with OnSolutionAdded for now.
-                string solutionPathToSetWithOnSolutionAdded = null;
-                var solution = (IVsSolution)Shell.ServiceProvider.GlobalProvider.GetService(typeof(SVsSolution));
-                if (solution != null && ErrorHandler.Succeeded(solution.GetSolutionInfo(out _, out var solutionFilePath, out _)))
+                // If we don't have any projects and this is our first project being added, then we'll create a new SolutionId
+                if (w.CurrentSolution.ProjectIds.Count == 0)
```

Comment:
I seem to remember that workspace would have a solution with default GUID when no solution is opened in VS, but looking at code here, it suggests otherwise. Is this recently changed or I was just plainly wrong about the state of workspace in this scenario?

Reply:
It might have done that before, is that something you think you're relying on?

Reply:
No, just curious